packages/core/core-flows/src/cart/workflows/complete-cart.ts

```diff
@@ -88,14 +88,14 @@ export const completeCartWorkflow = createWorkflow(
         const allItems: any[] = []
         const allVariants: any[] = []
 
-      data.cart?.items?.forEach((item) => {
-        allItems.push({
-          id: item.id,
-          variant_id: item.variant_id,
-          quantity: item.quantity,
+        data.cart?.items?.forEach((item) => {
+          allItems.push({
+            id: item.id,
+            variant_id: item.variant_id,
+            quantity: item.quantity,
+          })
+          allVariants.push(item.variant)
         })
```

Comment:
nit

```suggestion
const allVariants: any[] = []
const allItems = data.cart?.items?.map((item) => {
  allVariants.push(item.variant)
  return {
    id: item.id,
    variant_id: item.variant_id,
    quantity: item.quantity,
  }
})
```

Reply:
is it possible to have duplicated variant ids here?

Reply:
Yes, possible. Converted it to an object.